src/instruction/field_access.rs

```diff
@@ -61,24 +61,37 @@ impl Instruction for FieldAccess {
 mod tests {
     use super::*;
     use crate::instance::ToObjectInstance;
+    use crate::jinko;
     use crate::parser::Construct;
     use crate::JkInt;
 
-    fn setup() -> Context {
-        let mut ctx = Context::new();
-
-        let inst = Construct::instruction("type Point(x: int, y: int); ")
-            .unwrap()
-            .1;
-        inst.execute(&mut ctx);
+    // fn setup() -> Context {
+    //     let mut ctx = Context::new();
+    //
+    //     let inst = Construct::instruction("type Point(x: int, y: int); ")
+    //         .unwrap()
+    //         .1;
+    //     inst.execute(&mut ctx);
+    //
+    //     let inst = Construct::instruction("func basic() -> Point { Point { x = 15, y = 14 }}")
+    //         .unwrap()
+    //         .1;
+    //     inst.execute(&mut ctx);
+    //
+    //     let inst = Construct::instruction("b = basic();").unwrap().1;
+    //     inst.execute(&mut ctx);
+    //
+    //     ctx
+    // }
 
-        let inst = Construct::instruction("func basic() -> Point { Point { x = 15, y = 14 }}")
-            .unwrap()
-            .1;
-        inst.execute(&mut ctx);
+    fn setup() -> Context {
+        let mut ctx = jinko! {
+            type Point(x: int, y:int);
+            func basic() -> Point { Point { x = 15, y = 14 }}
+            b = basic();
+        };
 
-        let inst = Construct::instruction("b = basic();").unwrap().1;
-        inst.execute(&mut ctx);
+        ctx.execute().unwrap();
```

Comment:
I think this is super super cool!!!! Thanks so much!

There's a tiny nitpick I have with this. To me, it would make more sense if the `jinko!` macro was also executing code and not just building the context. So ideally we could skip the last line

Reply:
For this I believe that there is some code in the REPL which could help. But we can also skip this for now if you don't feel like it and we'll add it in a later PR

Reply:
The macro is already executing, I just forgot to remove this line